exporters/ostream/include/opentelemetry/exporters/ostream/log_exporter.h

```diff
@@ -0,0 +1,129 @@
+#pragma once
+
+#include "opentelemetry/logs/log_record.h"
+#include "opentelemetry/nostd/type_traits.h"
+#include "opentelemetry/sdk/logs/exporter.h"
+#include "opentelemetry/version.h"
+
+#include <iostream>
+#include <map>
+#include <sstream>
+
+namespace nostd   = opentelemetry::nostd;
+namespace sdklogs = opentelemetry::sdk::logs;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace logs
+{
+/**
+ * The OStreamLogExporter exports logs through an ostream (default set to std::cout)
+ */
+class OStreamLogExporter final : public sdklogs::LogExporter
+{
+private:
+  void print_value(common::AttributeValue &value)
+  {
+
+    // Based off api/include/opentelemetry/common/attribute_value.h
+    switch (value.index())
+    {
+      case common::AttributeType::TYPE_BOOL:
+        sout_ << nostd::get<bool>(value);
+        break;
+      case common::AttributeType::TYPE_INT:
+        sout_ << nostd::get<int>(value);
+        break;
+      case common::AttributeType::TYPE_INT64:
+        sout_ << nostd::get<int64_t>(value);
+        break;
+      case common::AttributeType::TYPE_UINT:
+        sout_ << nostd::get<unsigned int>(value);
+        break;
+      case common::AttributeType::TYPE_UINT64:
+        sout_ << nostd::get<uint64_t>(value);
+        break;
+      case common::AttributeType::TYPE_DOUBLE:
+        sout_ << nostd::get<double>(value);
+        break;
+      case common::AttributeType::TYPE_STRING:
+      case common::AttributeType::TYPE_CSTRING:
+        sout_ << nostd::get<nostd::string_view>(value);
+        break;
+
+      /*** Need to support these? ***/
+      case common::AttributeType::TYPE_SPAN_BOOL:
+        sout_ << nostd::get<nostd::span<const bool>>(value);
+        break;
+      case common::AttributeType::TYPE_SPAN_INT:
+        sout_ << nostd::get<nostd::span<const int>>(value);
+        break;
+      case common::AttributeType::TYPE_SPAN_INT64:
+        sout_ << nostd::get<nostd::span<const int64_t>>(value);
+        break;
+      case common::AttributeType::TYPE_SPAN_UINT:
+        sout_ << nostd::get<nostd::span<const unsigned int>>(value);
+        break;
+      case common::AttributeType::TYPE_SPAN_UINT64:
+        sout_ << nostd::get<nostd::span<const uint64_t>>(value);
+        break;
+      case common::AttributeType::TYPE_SPAN_DOUBLE:
+        sout_ << nostd::get<nostd::span<const double>>(value);
+        break;
+      case common::AttributeType::TYPE_SPAN_STRING:
+        sout_ << nostd::get<nostd::span<const nostd::string_view>>(value);
+        break;
+        /******** Up to here ************/
+
+      default:
+        sout_ << "Invalid type";
+        break;
+    }
+  }
+
+  void printKV(nostd::string_view &key, common::AttributeValue &value)
+  {
+    if (firstKV)
+    {
+      firstKV = false;
```

Comment:
I would recommend not to use a member variable like firstKV for intermediate state of this method, printKV, better to pass in a mutable reference or make a helper class dedicated to printing that has its own state.